lib/player/player.js

```diff
@@ -736,6 +741,36 @@ shaka.player.Player.prototype.setPlaybackRate = function(rate) {
 
 
 /**
+ * @param {Object} restrictions An Object containing the restrictions
+ *     on video tracks that you wish to set. For example, passing
+ *     {'maxBandwidth': 700000, 'minBandwidth': 200000} will restrict
+ *     playback to video tracks with bandwidth between 700000 and 200000.
+ * @export
+ */
+shaka.player.Player.prototype.setRestrictions = function(restrictions) {
+  for (var key in restrictions) {
+    if (this.restrictions_.hasOwnProperty(key)) {
+      this.restrictions_[key] = restrictions[key];
+    }
+  }
+  this.videoSource_.setRestrictions(this.restrictions_);
+};
+
+
+/**
+ * @return {Object} restrictions Contains the current restrictions
+ * @export
+ */
+shaka.player.Player.prototype.getRestrictions = function() {
```

Comment:
I like that this is a clone of the Restrictions object, but instead of returning a generic Object, let's return an instance of Restrictions instead.  To this end, please add a clone() method to Restrictions.prototype.